caliban_toolbox/dataset_builder.py

```diff
@@ -0,0 +1,627 @@
+        platform_list = np.array(platform_list)[combined_idx]
+
+        subset_dict = {'X': X, 'y': y, 'tissue_list': list(tissue_list),
+                       'platform_list': list(platform_list)}
+        return subset_dict
+
+    def _reshape_dict(self, data_dict, resize=False, output_shape=(512, 512), resize_target=400,
+                      resize_tolerance=1.5):
+        """Takes a dictionary of training data and reshapes it to appropriate size
+
+        data_dict: dictionary of training data
+        resize: flag to control resizing of the data.
+            Valid arguments:
+                    - False. No resizing
+                    - int/float: resizes by given ratio for all images
+                    - by_tissue. Resizes by median cell size within each tissue type
+                    - by_image. Resizes by median cell size within each image
+        output_shape: output shape for image data
+        resize_target: desired median cell size after resizing
+        resize_tolerance: sets maximum allowable ratio between resize_target and
+            median cell size before resizing occurs
+        """
+        X, y = data_dict['X'], data_dict['y']
+        tissue_list = np.array(data_dict['tissue_list'])
+        platform_list = np.array(data_dict['platform_list'])
+
+        if not resize:
+            # no resizing
+            X_new, y_new = reshape_training_data(X_data=X, y_data=y, resize_ratio=1,
+                                                 final_size=output_shape, stride_ratio=1)
+
+            # to preserve category labels, we need to figure out how much the array grew by
+            multiplier = int(X_new.shape[0] / X.shape[0])
+
+            # then we duplicate the labels in place to match expanded array size
+            tissue_list_new = [item for item in tissue_list for _ in range(multiplier)]
+            platform_list_new = [item for item in platform_list for _ in range(multiplier)]
+
+        elif isinstance(resize, (float, int)):
+            # resized based on supplied value
+            X_new, y_new = reshape_training_data(X_data=X, y_data=y, resize_ratio=resize,
+                                                 final_size=output_shape, stride_ratio=1,
+                                                 tolerance=resize_tolerance)
+
+            # to preserve category labels, we need to figure out how much the array grew by
+            multiplier = int(X_new.shape[0] / X.shape[0])
+
+            # then we duplicate the labels in place to match expanded array size
+            tissue_list_new = [item for item in tissue_list for _ in range(multiplier)]
+            platform_list_new = [item for item in platform_list for _ in range(multiplier)]
+        else:
+            X_new, y_new, tissue_list_new, platform_list_new = [], [], [], []
+
+            if resize == 'by_tissue':
+                batch_ids = np.unique(tissue_list)
+            else:
+                batch_ids = np.arange(0, X.shape[0])
```

Comment:
Even though the input gets validated in a different file, I would feel better if we specifically checked the string value instead of inferring it in the else.

```suggestion
            if resize == 'by_tissue':
                batch_ids = np.unique(tissue_list)
            elif resize == 'by_image':
                batch_ids = np.arange(0, X.shape[0])
            else:
                raise ValueError('Invalid `resize` value: {}'.format(resize))
```